test/utils/floating_point_utils.hpp

```diff
@@ -42,21 +53,46 @@ struct FP
     /// See: https://webassembly.github.io/spec/core/syntax/values.html#canonical-nan.
     static constexpr auto canon = UintType{1} << (num_mantissa_bits - 1);
 
-    T value{};
+private:
```

Comment:
Nit: better make it a class then and move `m_storage` to the top.

Reply:
Not really possible because `UintType` must be defined first.